tasks/request_files/request_files.py

```diff
@@ -138,19 +137,28 @@ def task(
         )
         retry_sleep_secs = DEFAULT_RESTORE_RETRY_SLEEP_SECS
 
-    # Get retrieval type
-    try:
-        retrieval_type = os.environ[OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY]
-        if retrieval_type not in ("Standard", "Bulk", "Expedited"):
-            msg = (
-                f"Invalid RESTORE_RETRIEVAL_TYPE: '{retrieval_type}'"
-                " defaulting to 'Standard'"
-            )
-            LOGGER.info(msg)
-            retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
-    except KeyError:
-        LOGGER.warn("Invalid RESTORE_RETRIEVAL_TYPE: 'None' defaulting to 'Standard'")
-        retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
+
+    VALID_RESTORE_TYPES = ["Bulk", "Expedited", "Standard"] 
+
+    # Get config and env set to None if the key does not exist
+    env_retrieval_type = os.getenv(OS_ENVIRON_RESTORE_DEFAULT_RECOVERY_TYPE_KEY, None)
+    config_retrieval_type = event["config"].get(CONFIG_RESTORE_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY, None)
+
+    # Set initial default to deployment env value
+    retrieval_type = env_retrieval_type
+
+    # Overide with config if we are able
+    if config_retrieval_type is not None:
+        if config_retrieval_type in VALID_RESTORE_TYPES:
+            retrieval_type = config_retrieval_type
+            LOGGER.info(f"Found glacier restore type from config: {retrieval_type}. Overridding other values.")
+        else:
+            LOGGER.error(f"Invalid restore type from config: {config_retrieval_type}. Using environment variable")
+        
+    # Safety catch in case someone put an invalid value in env
+    if (retrieval_type is None) or (retrieval_type not in VALID_RESTORE_TYPES):
+        LOGGER.info(f"invalid value for {retrieval_type}. Defaulting to {DEFAULT_RESTORE_RECOVERY_TYPE}")
```

Comment:
I would have this be a `error` instead of `info` logging type. Possibly add something like `check your {OS_ENVIRON_RESTORE_DEFAULT_RECOVERY_TYPE_KEY} value`

Reply:
updated